src/Compilers/CSharp/Test/Semantic/Semantics/DelegateTypeTests.cs

```diff
@@ -10490,6 +10490,187 @@ public void SynthesizedDelegateTypes_NamedArguments_MoreThan16Parameters()
+                    Report(lam2);
+                    var lam3 = (byte* a, int[] bs) => { };
+                    Report(lam3);
+                }
+                """;
+            CompileAndVerify(source, options: TestOptions.UnsafeReleaseExe, expectedOutput: $"""
+                <>f__AnonymousDelegate0
+                <>f__AnonymousDelegate0
+                <>f__AnonymousDelegate1
+                """).VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void SynthesizedDelegateTypes_DefaultParameterValues_DefaultUnification()
+        {
+            var source = """
+                using System;
+                static void Report(Delegate d) => Console.WriteLine(d.GetType());
+                var lam1 = (object o = null) => { };
+                Report(lam1);
+                var lam2 = (string s = null) => { };
+                Report(lam2);
+                var lam3 = (int? i = default) => { };
+                Report(lam3);
+                var lam4 = (S1 s1 = default) => { };
+                Report(lam4);
+                var lam5 = (S2 s2 = default) => { };
+                Report(lam5);
+                var lam6 = (int i = default) => { };
+                Report(lam6);
+
+                public struct S1 { public int X; }
+                public struct S2 { public int Y; }
+                """;
+            CompileAndVerify(source, expectedOutput: $"""
+                <>f__AnonymousDelegate0`1[System.Object]
+                <>f__AnonymousDelegate0`1[System.String]
+                <>f__AnonymousDelegate0`1[System.Nullable`1[System.Int32]]
+                <>f__AnonymousDelegate0`1[S1]
+                <>f__AnonymousDelegate0`1[S2]
+                <>f__AnonymousDelegate1`1[System.Int32]
+                """).VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void SynthesizedDelegateTypes_DefaultParameterValues_IntNullableUnification()
+        {
+            var source = """
+                using System;
+                static void Report(Delegate d) => Console.WriteLine(d.GetType());
+                var lam1 = (int a = 1) => { };
+                Report(lam1);
+                var lam2 = (int? b = 1) => { };
+                Report(lam2);
+                """;
+            CompileAndVerify(source, expectedOutput: $"""
+                <>f__AnonymousDelegate0`1[System.Int32]
+                <>f__AnonymousDelegate0`1[System.Nullable`1[System.Int32]]
+                """).VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void SynthesizedDelegateTypes_DefaultParameterValues_StringUnification()
+        {
+            var source = """
+                using System;
+                static void Report(Delegate d) => Console.WriteLine(d.GetType());
+                var lam1 = (string a = "") => { };
+                Report(lam1);
+                var lam2 = (string b = null) => { };
+                Report(lam2);
+                var lam3 = (string c = "abc") => { };
```

Comment:
>"abc" [](http://example.com/codeflow?start=39&length=5)

Can non null string default unify between different parameter types? Otherwise, adding a generic parameter type for a string with non-null default feels like an overkill.

Reply:
In metadata, types of default values don't have to match parameter types. At least in that case, non-null string default can unify between different parameter types. Let's demonstrate that.

Reply:
Thanks, demonstrated via `DefaultParameterValueAttribute`.